_plugins/editable_docs.rb

```diff
@@ -0,0 +1,55 @@
+Jekyll::Hooks.register :site, :pre_render do |site, payload|
+  repositories = site.data["remotes"]["repositories"]
+  site.pages.each do |page|
+    page_path = page.path
+    if page.data.key?("origin_path")
+      page_path = page.data["origin_path"]
+    end
+    next unless page_path.start_with?("doc")
+    repo_name, repo_data = repositories.find do |name, data|
+      page_path.start_with? (File.join("doc", name))
+    end
+    next if repo_name.nil? or repo_data.nil?
+
+    # Generate edit url
+    page_relative_url = page_path.sub(File.join("doc", repo_name), "")
+    page.data["edit_url"] = generate_edit_url(repo_data, page_relative_url)
+  end
+end
+
+def generate_edit_url(repo_data, page_relative_url)
+  https = repo_data['url'].include? "https"
+  github = repo_data['url'].include? "github.com"
+  bitbucket = repo_data['url'].include? "bitbucket.org"
+
+  unless github or bitbucket
+    raise ValueError("Cannot generate edition URL. Unknown organization for repository: #{repo_data['url']}")
+  end
+  
+  if https
```

Comment:
@apojomovsky I see that there's a logic to deal https vs. ssh and another logic to deal with git vs. bitbucket. Consider decoupling them to avoid code duplication.

Reply:
you're right, just did as suggested. thanks!